### PR TITLE
Updated information and code for App Router

### PR DESCRIPTION
**Pull request by royanger**
- Clarified api/ for Page Router and app/ for App Router
- Updated App Router example to remove unused imports, improve flow

### CODE REVIEW

File: docs/users/sync-data.mdx [hunk 1]
```diff
@@ -63,7 +63,7 @@ Now you need to set up your backend endpoint to receive the webhook events. The
 
 <Tab>
 
-If you're using Next.js, you can create a webhook endpoint in the `pages/api` directory.
+If you're using Next.js with Page Router, you can create a webhook endpoint in the `pages/api` directory.
```
Comment (alexisintech) [inline docs/users/sync-data.mdx:66]:
```suggestion
If you're using Next.js with Pages Router, you can create a webhook endpoint in the `pages/api` directory.
```

File: docs/users/sync-data.mdx [hunk 2]
```diff
@@ -150,57 +153,66 @@ Next you will want to retrieve the Webhook signing secret from the Clerk Dashboa
...
+```ts copy filename="app/api/webhook/route.ts"
+import { Webhook } from 'svix'
+import { headers } from 'next/headers'
+import { WebhookEvent } from '@clerk/nextjs/server'
 
 export async function POST(req: Request) {
-  const payload = await req.json();
-  const payloadString = JSON.stringify(payload);
+
+  // you can find this in the Clerk Dashboard -> Webhooks -> choose the webhook
```
Comment (alexisintech) [inline docs/users/sync-data.mdx:163]:
```suggestion
  // You can find this in the Clerk Dashboard -> Webhooks -> choose the webhook
```

File: docs/users/sync-data.mdx [hunk 3]
```diff
@@ -150,57 +153,66 @@ Next you will want to retrieve the Webhook signing secret from the Clerk Dashboa
...
+  const WEBHOOK_SECRET = process.env.WEBHOOK_SECRET
+
+  if (!WEBHOOK_SECRET) {
+    throw new Error('Please add WEBHOOK_SECRET from Clerk Dashboard to .env or .env.local')
+  }
+
+  // get the headers
```
Comment (alexisintech) [inline docs/users/sync-data.mdx:170]:
```suggestion
  // Get the headers
```

File: docs/users/sync-data.mdx [hunk 4]
```diff
@@ -150,57 +153,66 @@ Next you will want to retrieve the Webhook signing secret from the Clerk Dashboa
...
-  const svixId = headerPayload.get('svix-id');
-  const svixIdTimeStamp = headerPayload.get('svix-timestamp');
-  const svixSignature = headerPayload.get('svix-signature');
-  if (!svixId || !svixIdTimeStamp || !svixSignature) {
-    return new Response('Error occured', {
+  const svix_id = headerPayload.get("svix-id");
+  const svix_timestamp = headerPayload.get("svix-timestamp");
+  const svix_signature = headerPayload.get("svix-signature");
+
+  // if there are no headers, error out
```
Comment (alexisintech) [inline docs/users/sync-data.mdx:176]:
```suggestion
  // If there are no headers, error out
```

File: docs/users/sync-data.mdx [hunk 5]
```diff
@@ -150,57 +153,66 @@ Next you will want to retrieve the Webhook signing secret from the Clerk Dashboa
...
-  const svixHeaders = {
-    'svix-id': svixId,
-    'svix-timestamp': svixIdTimeStamp,
-    'svix-signature': svixSignature
-  };
-  // Create a new Webhook instance with your webhook secret
-  const wh = new Webhook(webhookSecret);
 
-  let evt: WebhookEvent;
+  // get the body
```
Comment (alexisintech) [inline docs/users/sync-data.mdx:183]:
```suggestion
  // Get the body
```

File: docs/users/sync-data.mdx [hunk 6]
```diff
@@ -150,57 +153,66 @@ Next you will want to retrieve the Webhook signing secret from the Clerk Dashboa
...
+  const payload = await req.json()
+  const body = JSON.stringify(payload);
+
+  // create a new SVIX instance with your secret.
```
Comment (alexisintech) [inline docs/users/sync-data.mdx:187]:
```suggestion
  // Create a new SVIX instance with your secret.
```

File: docs/users/sync-data.mdx [hunk 7]
```diff
@@ -150,57 +153,66 @@ Next you will want to retrieve the Webhook signing secret from the Clerk Dashboa
...
+  const wh = new Webhook(WEBHOOK_SECRET);
+
+  let evt: WebhookEvent
+
+  // verify the payload with the headers
```
Comment (alexisintech) [inline docs/users/sync-data.mdx:192]:
```suggestion
  // Verify the payload with the headers
```

File: docs/users/sync-data.mdx [hunk 8]
```diff
@@ -150,57 +153,66 @@ Next you will want to retrieve the Webhook signing secret from the Clerk Dashboa
...
+    }) as WebhookEvent
+  } catch (err) {
+    console.error('Error verifying webhook:', err);
     return new Response('Error occured', {
       status: 400
-    });
+    })
   }
+
+  // get the id and type
```
Comment (alexisintech) [inline docs/users/sync-data.mdx:206]:
```suggestion
  // Get the ID and type
```

**Review by alexisintech — CHANGES_REQUESTED** [review]

**Review by alexisintech — APPROVED** [review]
Last thing is that I would capitalize the comments for the App Router example as well, for consistency sake.
I'll go ahead and approve so you can make those changes and be all set 😸💖

**Comment by royanger** [comment]
Pretty sure you meant 'Pages Router example as well' but wanted to be clear on that before merging @alexisintech? Thanks for approving.

**Comment by alexisintech** [comment]
> Pretty sure you meant 'Pages Router example as well' but wanted to be clear on that before merging @alexisintech? Thanks for approving.

hmm, looking at the preview, the comments for the pages router example are capitalized. and the app router ones are not.

Uploading Screen Recording 2023-09-06 at 19.58.58.mov…

**Comment by royanger** [comment]
The deploy-preview didn't build the changes. Looking at the file in the most recent commit and I see all comments for all code examples capitalized. I can run the deploy-preview again if needed. Let me know.

**Comment by alexisintech** [comment]
> The deploy-preview didn't build the changes. Looking at the file in the most recent commit and I see all comments for all code examples capitalized. I can run the deploy-preview again if needed. Let me know.

oh then it's all good! I approved so you could use your better judgement, so go ahead and merge away if you feel ready 😄 it all looks good to me!

**Pull request merged**